### PR TITLE
Pass headers with sendFile request

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

File: main.coffee [hunk 1]
```diff
@@ -240,7 +240,16 @@ class JsonClient
                 else
                     form.append "file#{index}", file
 
-        form.submit url.resolve(@host, path), (err, res) ->
+        # build form data options from @options
+        options = buildOptions @options, @headers, @host, path, method: 'POST'
+        delete options.headers['content-type']
```
Comment (jsilvestre) [inline main.coffee]: why do you remove content-type?

File: main.coffee [hunk 2]
```diff
@@ -240,7 +240,16 @@ class JsonClient
...
+        if options.requestFactory is https
```
Comment (jsilvestre) [inline main.coffee]: `https` seems to be undefined

File: main.coffee (hunk 1 above)
Comment (aenario) [inline main.coffee]: because the form will have a "multipart" content-type. Added comment about it.

**Comment by jsilvestre** [comment]
You could remove build for 0.8 and add build for 0.11 with allowed_failure. What do you think?

**Comment by aenario** [comment]
doing this now.

@frankrousseau : do you want to support node 0.8 with request-json-light ?

**Comment by frankrousseau** [comment]
@aenario no :)

**Pull request merged**